src/librustc_span/lib.rs

```diff
@@ -1185,7 +1185,10 @@ impl SourceFile {
             *self.external_src.borrow(),
             ExternalSource::Foreign { kind: ExternalSourceKind::AbsentOk, .. }
         ) {
-            let src = get_src();
+            let src = get_src().map(|mut src| {
+                normalize_src(&mut src, BytePos::from_usize(0));
+                src
+            });
```

Comment:
Can you move this inside `if self.src_hash.matches(&src) {`? IIRC `src_hash` is computed pre-normalization, so we should also check it here in the same way.

Reply:
Good catch, it's pretty subtle. We really do need a test for this.

Reply:
There are some tests that print snippets from other crates, I think you might be able to do a search like `rg auxiliary/ src/test/**.stderr` to find some examples.

You can then combine that with an unnormalized file, maybe we have some normalization tests?

Reply:
That is, I think we can use an error message to print a snippet and trigger this through `rustc` instead of `rustdoc`.

Reply:
Oops, just saw your suggestion about how to do the test. I added one as a SourceMap test. Let me know if you'd prefer a integration-style test like you suggested, or if this is sufficient.

Reply:
`run-make` tests are a PITA, and your test is pretty clear, so no worries.